### ISSUE TITLE
Disallow defineExpose before any top-level await

### ISSUE THREAD

**Issue by AndreasNasman** [vuejs/eslint-plugin-vue#1872]
**Please describe what the rule should do:**
The rule should notify when `defineExpose` is used before any top-level `await`. Doing so causes problems as described in this core Vue.js issue:
https://github.com/vuejs/core/issues/4930#issuecomment-966544809

**What category should the rule belong to?**
[x] Warns about a potential error (problem)

**Provide 2-3 code examples that this rule should warn about:**

```vue
// Example 1
...
<script setup>
  ...
  const foo = { bar: 'baz' };
  await asyncOperation();
  defineExpose({ foo }); // `defineExpose` should be called before any top-level `await`.
  ...
</script>
...

// Example 2
...
<script setup>
  ...
  await asyncOperation();
  const foo = { bar: 'baz' };
  defineExpose({ foo }); // `defineExpose` should be called before any top-level `await`.
  ...
</script>
...
```

**Additional context**
The current workaround is to manually remember to place all `await`s after `defineExpose`. Forgetting to do so leads to problems that are hard to debug. In the context of my examples, a JavaScript error would be thrown when `foo` is used (`foo is not defined` or similar) in some other component and it's unclear the source of the problem is actually in the component defining `foo`.

The problem seems challenging to fix in Vue.js itself:
https://github.com/vuejs/core/issues/4930#issuecomment-966544809

**Comment by AndreasNasman** [comment]
To clarify, this only happened in a production build – the development build worked fine – and was therefore extra hard to debug.

**Comment by FloEdelmann** [comment]
The rule [`vue/no-expose-after-await`](https://eslint.vuejs.org/rules/no-expose-after-await.html) should already catch this case. It is currently uncategorized and has to be enabled manually. It will be added to all Vue 3 preset configs in the next major version: #1848

Could you please enable it manually and check whether it reports all problematic cases?

**Comment by AndreasNasman** [comment]
I scaffolded a new Vue project with `npm init vue@latest` and it seems to only work for regular `script` components taking `expose` as a `setup` parameter, not `script setup` that uses `defineExpose`.

<img width="1424" alt="image" src="https://user-images.githubusercontent.com/17124245/165728680-f6a53136-b73d-43ea-bfbc-2904dd9dab45.png">

I upgraded the ESLint and eslint-plugin-vue versions to the latest stable releases, 8.14.0 and 8.7.1 respectively, so it's present in the latest stable versions.

**Comment by FloEdelmann** [comment]
Thanks for testing! So the [vue/no-expose-after-await](https://eslint.vuejs.org/rules/no-expose-after-await.html) rule should be adapted to also catch cases in `<script setup>`.
